zrml/prediction-markets/src/migrations.rs

```diff
@@ -15,6 +15,427 @@
+                    })
+                } else {
+                    None
+                };
+                let oracle = Some(Bond {
+                    who: old_market.creator.clone(),
+                    value: T::OracleBond::get(),
+                    is_settled: matches!(
+                        old_market.status,
+                        MarketStatus::Resolved | MarketStatus::InsufficientSubsidy,
+                    ),
+                });
+                let validity = if old_market.creation == MarketCreation::Permissionless {
+                    Some(Bond {
+                        who: old_market.creator.clone(),
+                        value: T::ValidityBond::get(),
+                        is_settled: matches!(
+                            old_market.status,
+                            MarketStatus::Resolved | MarketStatus::InsufficientSubsidy,
+                        ),
+                    })
+                } else {
+                    None
+                };
+                let new_market = Market {
+                    creator: old_market.creator,
+                    creation: old_market.creation,
+                    creator_fee: old_market.creator_fee,
+                    oracle: old_market.oracle,
+                    metadata: old_market.metadata,
+                    market_type: old_market.market_type,
+                    period: old_market.period,
+                    scoring_rule: old_market.scoring_rule,
+                    status: old_market.status,
+                    report: old_market.report,
+                    resolved_outcome: old_market.resolved_outcome,
+                    dispute_mechanism: old_market.dispute_mechanism,
+                    deadlines: old_market.deadlines,
+                    bonds: MarketBonds { advisory, oracle, validity },
+                };
+                (key, new_market)
+            })
+            .collect::<Vec<_>>();
+
+        for (key, new_market) in new_markets {
+            put_storage_value::<MarketOf<T>>(MARKET_COMMONS, MARKETS, &key, new_market);
+            total_weight = total_weight.saturating_add(T::DbWeight::get().writes(1));
+        }
+
+        StorageVersion::new(MARKET_COMMONS_NEXT_STORAGE_VERSION).put::<MarketCommonsPallet<T>>();
+        total_weight = total_weight.saturating_add(T::DbWeight::get().writes(1));
+        log::info!("RecordBonds: Done!");
+        total_weight
+    }
+
+    #[cfg(feature = "try-runtime")]
+    fn pre_upgrade() -> Result<(), &'static str> {
+        let old_markets =
+            storage_iter::<OldMarketOf<T>>(MARKET_COMMONS, MARKETS).collect::<BTreeMap<_, _>>();
+        Self::set_temp_storage(old_markets, "old_markets");
+        Ok(())
+    }
+
+    #[cfg(feature = "try-runtime")]
+    fn post_upgrade() -> Result<(), &'static str> {
+        let old_markets: BTreeMap<MarketIdOf<T>, MarketOf<T>> =
+            Self::get_temp_storage("old_markets").unwrap();
```

Comment:
The `make try-runtime-upgrade-zeitgeist` command fails here:
> Thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', /home/sea212/Documents/github/zeitgeist/zrml/prediction-markets/src/migrations.rs:166
>
>This is a bug. Please report it at:
>
>        https://github.com/zeitgeistpm/zeitgeist/issues
>
>Error: Input("failed to execute TryRuntime_on_runtime_upgrade: Runtime panicked: called `Option::unwrap()` on a `None` value")

There a couple of things that cause problems here:
1. in `post_upgrade`, when fetching the old markets the [old market data is mapped to the new market structure](https://github.com/zeitgeistpm/zeitgeist/pull/892/files#diff-7eef70e0e24e29abbdd8ae4c5949b3d7633f1fa7b844e9b0e7ff3227cc344865R163-R164).
2. [Old markets are stored using a `Vec<u8>`](https://github.com/zeitgeistpm/zeitgeist/pull/892/files#diff-7eef70e0e24e29abbdd8ae4c5949b3d7633f1fa7b844e9b0e7ff3227cc344865R156) that represents the encoded key, but later they are [tried to be retrieved using a `MarketIdOf<T>`](https://github.com/zeitgeistpm/zeitgeist/pull/892/files#diff-7eef70e0e24e29abbdd8ae4c5949b3d7633f1fa7b844e9b0e7ff3227cc344865R167-R169).

Good news is, the `on_runtime_upgrade` logic works as expected :+1:

```suggestion
    #[cfg(feature = "try-runtime")]
    fn pre_upgrade() -> Result<(), &'static str> {
        use frame_support::{migration::storage_key_iter, pallet_prelude::Blake2_128Concat};

        let old_markets = storage_key_iter::<MarketIdOf<T>, OldMarketOf<T>, Blake2_128Concat>(
            MARKET_COMMONS,
            MARKETS,
        )
        .collect::<BTreeMap<_, _>>();
        Self::set_temp_storage(old_markets, "old_markets");
        Ok(())
    }

    #[cfg(feature = "try-runtime")]
    fn post_upgrade() -> Result<(), &'static str> {
        let old_markets: BTreeMap<MarketIdOf<T>, OldMarketOf<T>> =
            Self::get_temp_storage("old_markets").unwrap();
```

Reply:
Thanks. :thumbsup: I didn't even get to run `try-runtime` because the update wasn't deployed yet.